### ISSUE TITLE
OpenBSD 7.4 - shutdown hangs (only x86_64, not arm64)

### ISSUE THREAD

**Issue by kobalicek** [cross-platform-actions/action#72]
I'm still having shutdown failures now with OpenBSD 7.4 VM:

```
  Shuting down VM
  Executing command inside VM: sudo shutdown -h -p now
  /usr/bin/ssh -t runner@192.168.64.2
  Pseudo-terminal will not be allocated because stdin is not a terminal.
  Error: The operation was canceled.
```

Here is a reference:
  
  https://github.com/blend2d/blend2d/actions/runs/6763854194/job/18381464122

Note that this must be a deadlock or something because the runner is killed in 6 hours. The workflow is using `sync_files: runner-to-vm`, but that doesn't fix the problem.

One interesting thing is that this doesn't happen with ARM64 VM, only x86_64 one.

So, opening this issue to see whether there are others having the same problem.

**Comment by jacob-carlborg** [comment]
I've added some more debug information and it seems to stop making progress in the SSH command to shut down the VM. Specifically, just before authentication is supposed to succeed.

**Comment by opcm** [comment]
same here: https://github.com/intel/pcm/actions/runs/7128218686/job/19443586521

**Comment by jacob-carlborg** [comment]
> same here: https://github.com/intel/pcm/actions/runs/7128218686/job/19443586521

That's for FreeBSD, so it's not just a problem with OpenBSD. It might be an issue with xhyve since it doesn't seem to be a problem on ARM64. QEMU is used on ARM64. 

It might also not be just be a problem with shutting down the VM. Here's one of my projects that stops making progress in the middle of the job, not at the end: https://github.com/jacob-carlborg/dstep/actions/runs/7040324159/job/19161045455#step:6:429.

I think I can specify a timeout for the SSH commands that are used to communicate with the VM. Combining that with a retry might help. Or at least kill the job earlier.

**Comment by jacob-carlborg** [comment]
Hopefully fixed/mitigated in 01856ae87cf2d91d4fe7f3933180d252d94f3c07 and d9184930c333266f6819290a62817f3dc884ffd5.
